test/common.js

```diff
@@ -0,0 +1,19 @@
+var yaml = require('js-yaml');
+
+module.exports.getDiag = function (body) {
+    var yamlStart = body.indexOf('  ---');
+    var yamlEnd = body.indexOf('  ...\n');
+    var diag = body.slice(yamlStart, yamlEnd).split('\n').map(function (line) {
+        return line.slice(2);
+    }).join('\n');
+
+    // The stack trace will vary depending on where the code is run, so just
+    // strip it out.
+    var withStack = yaml.safeLoad(diag);
+    delete withStack.stack;
+    return withStack;
+}
+
+module.exports.stripFullStack = function (output) {
+    return output.replace(/^\W+at.*:\d+:\d+.*$\W/gm, '');
```

Comment:
Re 1, we can determine the filename with `__filename` and `process.cwd()` and similar.
Changes to line position are a fair point, and i'd be fine normalizing all of those (ie, all line numbers would be normalized to 0 - so that relative line numbers mattered but not absolute numbers).

I guess I'm mainly concerned (especially because this regex is pretty inscrutable) that there will be bugs in the stack traces, and we won't know.